surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/RunListenerAdapter.java

```diff
@@ -173,8 +175,9 @@ private SimpleReportEntry createReportEntry( TestIdentifier testIdentifier,
         {
             classText = null;
         }
-        String methodName = testIdentifier.isTest() ? classMethodName[2] : null;
-        String methodText = testIdentifier.isTest() ? classMethodName[3] : null;
+        boolean isFailedContainer = isFailedContainer( testIdentifier, testExecutionResult );
+        String methodName = ( isFailedContainer || testIdentifier.isTest() ) ? classMethodName[2] : null;
```

Comment:
You do not have to use brackets in `( isFailedContainer || testIdentifier.isTest() )`. There is no collision between boolean and String in this case.